src/clr-angular/forms/select/options.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component, ElementRef, HostBinding, Inject, Injector, OnDestroy, Optional} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {AbstractPopover} from "../../popover/common/abstract-popover";
+import {Point} from "../../popover/common/popover";
+import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+@Component({selector: "clr-options", templateUrl: "options.html", host: {"[class.clr-options]": "true"}})
+export class ClrOptions extends AbstractPopover implements OnDestroy {
+    // Have to declare this separately or TS complains that super call in the constructor
+    // should be before the if condition which we do not want in this case
+    protected ifOpenService: IfOpenService;
+    private sub: Subscription;
+
+    constructor(injector: Injector, @Optional() @Inject(POPOVER_HOST_ANCHOR) parentHost: ElementRef,
+                ifOpenService: IfOpenService) {
+        if (!parentHost) {
+            throw new Error("clr-select-menu should only be used inside of a clr-select");
+        }
+        super(injector, parentHost);
+
+        // Initialize Services
+        this.ifOpenService = ifOpenService;
+
+        // Configure Popover
+        this.initializeSubscriptions();
+        this.configurePopover();
+    }
+
+    @HostBinding("class.open")
+    get open(): boolean {
+        return this.ifOpenService.open;
+    }
+
+    /**
+     * Configure Popover Direction and Close indicators
+     */
+    private configurePopover(): void {
+        this.anchorPoint = Point.BOTTOM_LEFT;
+        this.popoverPoint = Point.LEFT_TOP;
+        this.closeOnOutsideClick = true;
+    }
+
+    private initializeSubscriptions(): void {
+        if (this.ifOpenService) {
+            this.sub = this.ifOpenService.ignoredElementChange.subscribe((el: ElementRef) => {
+                if (el) {
+                    this.ignoredElement = el.nativeElement;
+                }
+                this.sub.unsubscribe();
+            });
+        }
+    }
+
+    // Lifecycle hooks
+    ngOnDestroy() {
+        if (!this.sub.closed) {
```

Comment:
Shouldn't this be just `if(this.sub)`?

Reply:
no because I also unsubscribe on line 57. If that unsubscribe hasn't been triggered, only then should we unsubscribe here too.

Reply:
But what if this component gets destroyed before this `initializeSubscriptions()` method called? Also, if `this.ifOpenService` is `falsy`, then the subscription wouldn't be set. Then the `this.sub` will be `undefined` though, right?

Reply:
I dont understand. If the component gets destroyed before the `initializeSubscriptions` then the `Subscription` wont be initialized in the first place. Why would we need to unsubscribe in that case? And I don't think thats a valid use case or else we would have to check if the subscription is null or not before unsubscribing in all our components.

Reply:
If we are subscribing under a certain condition, we should check if the subscription property is truthy before calling its `unsubscribe()` method in `ngOnDestroy()`. Otherwise, that would throw an error as we try to call a method of `undefined`. In this case, we are subscribing under this condition, `if(this.ifOpenService)`.  And, I believe we actually do this in some of our components. I found `label.ts` does just that.

Reply:
wait are you talking about this condition? `if (this.ifOpenService) {`

If yes, yes I have to remove that condition. Its always going to be true.

Reply:
Yes, I was talking about that condition :D